src/NSwag.CodeGeneration/Models/ResponseModelBase.cs

```diff
@@ -86,8 +86,10 @@ public bool IsDate
 
         /// <summary>Gets a value indicating whether the response requires a text/plain content.</summary>
         public bool IsPlainText =>
-            !_response.Content.ContainsKey("application/json") &&
-            (_response.Content.ContainsKey("text/plain") || _operationModel.Produces == "text/plain");
+            (_response.Content.ContainsKey("application/json") &&
+            _response.Schema != null && _response.Schema?.Type == JsonObjectType.String) ||
```

Comment:
The problem is that this is actually a hack for ASP.NET Core but from the perspective of OAI not correct... i.e. a response of type JSON and string should return `"foo bar"` and not `foo bar` - the problem is that ASP.NET Core (at least in older versions) returned a string and not a JSON string changing the type of the response from JSON to text/plain.

Reply:
Ref: https://github.com/aspnet/Mvc/issues/4945

Reply:
I was not aware of that specific hack for ASP.NET Core and I see how this change would not be a sufficient fix. It seems that in some locations throughout the codebase preference is given to "application/json" over "text/plain" for response types. Would it then be sufficient to select a response type of "application/json" when there are both "application/json" and "text/plain" are available? This would be a special case for strings.